libwhisper/BloomFilter.h

```diff
@@ -83,12 +86,51 @@ void TopicBloomFilterBase<N>::removeRaw(FixedHash<N> const& _h)
 
 template <unsigned N>
 bool TopicBloomFilterBase<N>::isBitSet(FixedHash<N> const& _h, unsigned _index)
-{	
+{
 	unsigned iByte = _index / 8;
-	unsigned iBit = _index % 8;
+	unsigned iBit = _index & 0x7;
 	return (_h[iByte] & c_powerOfTwoBitMmask[iBit]) != 0;
 }
 
+template <unsigned N>
+void TopicBloomFilterBase<N>::setBit(FixedHash<N>& _h, unsigned _index)
+{
+	unsigned iByte = _index / 8;
+	unsigned iBit = _index & 0x7;
+	_h[iByte] |= c_powerOfTwoBitMmask[iBit];
+}
+
+template <unsigned N>
+FixedHash<N> TopicBloomFilterBase<N>::bloom(AbridgedTopic const& _h)
+{
+	// The size of AbridgedTopic is 32 bits, and 27 of them participate in this algorithm.
```

Comment:
`_h` is a cryptographic hash. All bits are independent and contain close to full entropy. There is no reason to utilise more bits than that required for placing the 1s in the bloom. I'm against using two algorithms for the same thing; blooms in Ethereum proper have had many more eyes and I consider them more secure.